### PR TITLE
Make sure the blogID passed to Stats service is not blank.

### LINKED ISSUE #2241: Crash Report 3.6: Unable to stop service StatsService
```
java.lang.RuntimeException: Unable to stop service org.wordpress.android.ui.stats.service.StatsService@42e31750: java.util.ConcurrentModificationException
       at android.app.ActivityThread.handleStopService(ActivityThread.java:3149)
       at android.app.ActivityThread.access$2300(ActivityThread.java:174)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1423)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:146)
       at android.app.ActivityThread.main(ActivityThread.java:5748)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:515)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1291)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1107)
       at dalvik.system.NativeStart.main(NativeStart.java)
Caused by: java.util.ConcurrentModificationException
       at java.util.LinkedList$LinkIterator.next(LinkedList.java:124)
       at org.wordpress.android.ui.stats.service.StatsService.stopRefresh(StatsService.java:135)
       at org.wordpress.android.ui.stats.service.StatsService.onDestroy(StatsService.java:68)
       at android.app.ActivityThread.handleStopService(ActivityThread.java:3132)
       at android.app.ActivityThread.access$2300(ActivityThread.java:174)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1423)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:146)
       at android.app.ActivityThread.main(ActivityThread.java:5748)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:515)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1291)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1107)
       at dalvik.system.NativeStart.main(NativeStart.java)
```

### PR DESCRIPTION
**Pull request by daniloercoli**
This PR fixes 2 rare bugs.

The first bug #2240 is a very rare crash, that happens where the blogId passed to the Service is null. Not sure how this can happen.

The second bug #2241 is another rare crash  that happens when 2 threads try to access the service. One to stop it, and the other to start it with new blogID, and timeperiod. Fixed by making sure to get the lock on the networkCallsList before accessing it.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/stats/service/StatsService.java [hunk 1]
```diff
@@ -88,7 +87,11 @@ public int onStartCommand(Intent intent, int flags, int startId) {
             return START_NOT_STICKY;
         }
 
-        final String blogId = StringUtils.notNullStr(intent.getStringExtra(ARG_BLOG_ID));
+        final String blogId = intent.getStringExtra(ARG_BLOG_ID);
+        if (org.apache.commons.lang.StringUtils.isBlank(blogId)) {
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/service/StatsService.java]: We should use `TextUtils.isEmpty` for consistency.

File: WordPress/src/main/java/org/wordpress/android/ui/stats/service/StatsService.java [hunk 2]
```diff
@@ -98,10 +101,15 @@ public int onStartCommand(Intent intent, int flags, int startId) {
...
-                    + mServiceBlogId + ". Using current date!!!");
-            int localTableBlogId = WordPress.wpDB.getLocalTableBlogIdForRemoteBlogId(
-                    Integer.parseInt(mServiceBlogId));
-            requestedDate = StatsUtils.getCurrentDateTZ(localTableBlogId);
+                    + blogId + ". Using current date!!!");
+            try {
+                int parsedBlogID = Integer.parseInt(blogId);
+                int localTableBlogId = WordPress.wpDB.getLocalTableBlogIdForRemoteBlogId(parsedBlogID);
+                requestedDate = StatsUtils.getCurrentDateTZ(localTableBlogId);
+            } catch (NumberFormatException e) {
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/service/StatsService.java]: Probably not necessary (I fear that could hide other errors).

File: WordPress/src/main/java/org/wordpress/android/ui/stats/service/StatsService.java [hunk 3]
```diff
@@ -128,18 +136,20 @@ public int onStartCommand(Intent intent, int flags, int startId) {
...
 
     private void stopRefresh() {
-        this.mServiceBlogId = null;
-        this.mServiceRequestedTimeframe = StatsTimeframe.DAY;
-        this.mServiceRequestedDate = null;
-        this.mServiceStartId = 0;
-        for (Request<JSONObject> req : statsNetworkRequests) {
-            if (req != null && !req.hasHadResponseDelivered() && !req.isCanceled()) {
-                req.cancel();
+        synchronized (statsNetworkRequests) {
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/service/StatsService.java]: We should rename `statsNetworkRequests` to `mStatsNetworkRequests` - I was confused while reading the diff, I thought it was loca (so no reason to synchronize it).

File: WordPress/src/main/java/org/wordpress/android/ui/stats/service/StatsService.java (hunk 1 above)
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/service/StatsService.java]: I switched to `isEmpty`, but the semantics of the 2 functions is a little bit different. `isBlank` returns `true` on a string that is a sequence of whitespace characters, but  `isEmpty` doesn't.

**Comment by maxme** [comment]
:shipit:

File: WordPress/src/main/java/org/wordpress/android/ui/stats/service/StatsService.java (hunk 1 above)
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/service/StatsService.java]: If we have a whitespace encoded blog id here, I'll be worried;)

**Pull request merged**